xbmc/pvr/PVRDatabase.cpp

```diff
@@ -289,7 +299,8 @@ bool CPVRDatabase::DeleteChannels()
   return DeleteValues("channels");
 }
 
-bool CPVRDatabase::Delete(const CPVRChannel& channel)
+// add bCommit here
+bool CPVRDatabase::Delete(const CPVRChannel& channel, bool bCommit)
```

Comment:
bCommit is a dummy; implementation is missing.

Will CPVRDatabase::Delete now ever be called with bCommit = true? If not, rename this method and implement only queue and no commit sematics.

Reply:
Same here